libs/community/tests/unit_tests/vectorstores/test_sqlserver.py

```diff
@@ -59,130 +73,114 @@ def texts() -> List[str]:
+@pytest.fixture
+def metadatas() -> List[dict]:
+    """Definition of metadatas used in the tests."""
+    query_metadata = [
+        {"id": 1, "summary": "Good Quality Dog Food"},
+        {"id": 2, "summary": "Nasty No flavor"},
+        {"id": 3, "summary": "stale product"},
+        {"id": 4, "summary": "Great value and convenient ramen"},
+        {"id": 5, "summary": "Great for the kids!"},
+    ]
+    return query_metadata  # provide this data to the test.
+
+
+@pytest.fixture
+def docs() -> List[Document]:
+    """Definition of doc variable used in the tests."""
+    docs = [
+        Document(
+            page_content="rabbit",
+            metadata={"color": "black", "type": "pet", "length": 6},
+        ),
+        Document(
+            page_content="cherry",
+            metadata={"color": "red", "type": "fruit", "length": 6},
+        ),
+        Document(
+            page_content="hamster",
+            metadata={"color": "brown", "type": "pet", "length": 7},
+        ),
+        Document(
+            page_content="cat", metadata={"color": "black", "type": "pet", "length": 3}
+        ),
+        Document(
+            page_content="elderberry",
+            metadata={"color": "blue", "type": "fruit", "length": 10},
+        ),
     ]
+    return docs  # provide this data to the test
+
+
+def test_sqlserver_add_texts(
+    store: SQLServer_VectorStore,
+    texts: List[str],
+    metadatas: List[dict],
+) -> None:
+    """Test that add text returns equivalent number of ids of input texts."""
     result = store.add_texts(texts, metadatas)
     assert len(result) == len(texts)
 
 
 def test_sqlserver_add_texts_when_no_metadata_is_provided(
     store: SQLServer_VectorStore,
+    texts: List[str],
 ) -> None:
     """Test that when user calls the add_texts function without providing metadata,
     the embedded text still get added to the vector store."""
-    texts = [
-        "Good review",
-        "new books",
-        "table",
-        "Sunglasses are a form of protective eyewear.",
-    ]
     result = store.add_texts(texts)
     assert len(result) == len(texts)
 
 
 def test_sqlserver_add_texts_when_text_length_and_metadata_length_vary(
     store: SQLServer_VectorStore,
+    texts: List[str],
+    metadatas: List[dict],
 ) -> None:
     """Test that all texts provided are added into the vector store
     even when metadata is not available for all the texts."""
-    # The text 'elderberry' and its embedded value should be added to the vector store.
-    texts = ["rabbit", "cherry", "hamster", "cat", "elderberry"]
-    metadatas = [
-        {"color": "black", "type": "pet", "length": 6},
-        {"color": "red", "type": "fruit", "length": 6},
-        {"color": "brown", "type": "pet", "length": 7},
-        {"color": "black", "type": "pet", "length": 3},
-    ]
+    # We get all metadatas except the last one from our metadatas fixture.
+    # The text without a corresponding metadata should be added to the vector store.
```

Comment:
why is there no metadata for the last text?

Reply:
The test case checks that if a user does not provide metadata for a text, the text still gets added to the store. To simulate this is why the last text does not have metadata, it’s just mimicking a sample user input.